go/pkg/client/cas.go

```diff
@@ -74,6 +74,32 @@ type uploadState struct {
 	cancel  func()
 }
 
+// StatusDetailsString returns the error details of the status as a text string.
+func StatusDetailsString(st *status.Status) string {
+	var details []string
+	for _, d := range st.Details() {
+		s := d.(string)
+		if pb, ok := d.(proto.Message); ok {
+			s = proto.MarshalTextString(pb)
+		}
+		details = append(details, s)
+	}
+	return strings.Join(details, "; ")
+}
+
+// StatusDetailedMessage is an extension of st.Err().Error() that also includes the error details of the status.
+func StatusDetailedMessage(st *status.Status) string {
```

Comment:
maybe `StatusDetailedError`?

Reply:
I haven't sent this out to review yet, because it doesn't actually work :-)
But yes, I made it an error on one of the revisions.

Reply:
Okay, ready now. Apparently, needed to add an unused import for the error details proto. Tested by deliberately triggering failed precondition errors.